File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/event/MyStartingEventListener.java
package io.micronaut.rabbitmq.docs.event;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.context.event.ApplicationEventListener;
import io.micronaut.rabbitmq.event.RabbitConsumerStarting;
import jakarta.inject.Singleton;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
// end::imports[]

@Requires(property = "spec.name", value = "RabbitListenerEventsSpec")
// tag::clazz[]
@Singleton
public class MyStartingEventListener implements ApplicationEventListener<RabbitConsumerStarting> {
  private static final Logger LOG = LoggerFactory.getLogger(MyStartingEventListener.class);
  @Override
  public void onApplicationEvent(RabbitConsumerStarting event) {
    LOG.info("RabbitMQ consumer: {} (method: {}) is subscribing to: {}",
      event.getSource(), event.getMethod(), event.getQueue());
  }
}
// end::clazz[]

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/event/MyStartingEventListener.py
from micronaut.context.annotation import Requires
# tag::imports[]
import logging

from jakarta.inject import Singleton
from micronaut.context.event import ApplicationEventListener
from micronaut.rabbitmq.event import RabbitConsumerStarting

LOG = logging.getLogger(__name__)
# end::imports[]


@Requires(property="spec.name", value="RabbitListenerEventsSpec")
# tag::clazz[]
@Singleton
class MyStartingEventListener(ApplicationEventListener[RabbitConsumerStarting]):

    def onApplicationEvent(self, event: RabbitConsumerStarting) -> None:
        LOG.info("RabbitMQ consumer: %s (method: %s) is subscribing to: %s",
                 event.getSource(), event.getMethod(), event.getQueue())
# end::clazz[]
